zephyr/include/sof/auth/auth_api_iface.h

```diff
@@ -0,0 +1,149 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+ *         Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+ */
+#ifndef __AUTH_API_IFACE_H__
+#define __AUTH_API_IFACE_H__
+
+#include "status_logger_iface.h"
+#include <stdint.h>
+#include <stddef.h>
+
+#define AUTH_API_VERSION_MAJOR (2)
+#define AUTH_API_VERSION_MINOR (0)
+#define AUTH_API_VERSION_PATCH (0)
+
+#define AUTH_SCRATCH_BUFF_SZ (0xA000) // 40kB
```

Comment:
From API it seems to be configurable, but I follow ROM_EXT definitions since we have to use size used during ROM_EXT build. The auth_api_init() input parameters are checked against values set in auth_api_iface.h.